From: Gianfranco Costamagna <locutusofborg@debian.org>
Date: Sun, 11 Sep 2016 22:00:11 +0200
Subject: Find and link libs

Fix build failures by detecting and linking libraries:
- OPENGL
- Threads
- EXPAT
---
 CMakeLists.txt     | 4 ++++
 src/CMakeLists.txt | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 03f3e47..9f18742 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,6 +139,9 @@ set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${MSVC_LD_FLAGS}")
 
 find_package(Boost REQUIRED)
 find_package(ZLIB REQUIRED)
+find_package(Threads REQUIRED)
+find_package(EXPAT REQUIRED)
+
 if (SHARED_FLTK)
     #see http://sources.debian.net/src/fltk1.3/1.3.2-4/README.CMake.txt lines 200-240
     find_package(FLTK REQUIRED NO_MODULE)
@@ -149,6 +152,7 @@ else (SHARED_FLTK)
 endif (SHARED_FLTK)
 find_package(OpenSceneGraph 2.8.1 REQUIRED osgText osgDB osgParticle osgUtil osgViewer osgGA)
 find_package(SimGear ${FGRUN_VERSION} REQUIRED)
+find_package(OpenGL REQUIRED)
 if (ENABLE_NLS)
     find_package(Gettext)
 endif (ENABLE_NLS)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index f32e1c2..b619f3b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -74,6 +74,8 @@ target_link_libraries(fgrun
     ${PLATFORM_LIBS}
     ${OPENGL_LIBRARIES}
     ${CMAKE_DL_LIBS}
+    ${CMAKE_THREAD_LIBS_INIT}
+    ${EXPAT_LIBRARIES}
     )
 
 install(TARGETS fgrun RUNTIME DESTINATION bin)
